src/Storage/Device/Local.php

```diff
@@ -48,67 +48,146 @@ public function getRoot(): string
 
     /**
      * @param string $filename
+     * @param string $prefix
      *
      * @return string
      */
-    public function getPath($filename): string
+    public function getPath(string $filename, string $prefix = null): string
     {
         $path = '';
 
         for ($i = 0; $i < 4; ++$i) {
             $path = ($i < \strlen($filename)) ? $path . DIRECTORY_SEPARATOR . $filename[$i] : $path . DIRECTORY_SEPARATOR . 'x';
         }
-
+        if(!is_null($prefix)) {
+            $path = $prefix . DIRECTORY_SEPARATOR . $path;
+        }
         return $this->getRoot() . $path . DIRECTORY_SEPARATOR . $filename;
     }
 
     /**
      * Upload.
      *
      * Upload a file to desired destination in the selected disk.
+     * return number of chunks uploaded or 0 if it fails.
      *
      * @param string $source
      * @param string $path
+     * @param int $chunk
+     * @param int $chunks
+     * @param array $metadata
      *
      * @throws \Exception
      *
-     * @return bool
+     * @return int
      */
-    public function upload($source, $path): bool
+    public function upload(string $source, string $path, int $chunk = 1, int $chunks = 1, array &$metadata = []): int
     {
         if (!\file_exists(\dirname($path))) { // Checks if directory path to file exists
             if (!@\mkdir(\dirname($path), 0755, true)) {
                 throw new Exception('Can\'t create directory: ' . \dirname($path));
             }
         }
 
-        if (\move_uploaded_file($source, $path)) {
-            return true;
+        //move_uploaded_file() verifies the file is not tampered with
+        if($chunks === 1) {
+            if (!\move_uploaded_file($source, $path)) {
+                throw new Exception('Can\'t upload file ' . $path);
+            }
+            return $chunks;
         }
+        $tmp = \dirname($path) . '/tmp/' . \basename($path) . '_chunks.log';
```

Comment:
Don't we need to use `DIRECTORY_SEPARATOR` instead of `/` here?

Reply:
Good idea!